src/inference/smc.js

```diff
@@ -145,7 +159,10 @@ module.exports = function(env) {
     var logW = util.logsumexp(_.map(particles, 'logWeight'));
     var logAvgW = logW - Math.log(m);
 
-    assert.notStrictEqual(logAvgW, -Infinity, 'All particles have zero weight.');
+    if (logAvgW === -Infinity) {
+      // do not return, execution continues
+      env.coroutine.error('All particles have zero weight.');
+    }
```

Comment:
Like this?
```
  SMC.prototype.sync = function() {
    // Called at sync points factor and exit.
    // Either advance the next active particle, or if all particles have
    // advanced, perform re-sampling and rejuvenation.
    if (!this.atLastParticle()) {
      this.advanceParticleIndex();
      return this.runCurrentParticle();
    } else {
      this.step += 1;
      this.debugLog('***** sync :: step = ' + this.step + ' *****');

      // Resampling and rejuvenation are applied to all particles.
      // Active and complete particles are combined here and
      // re-partitioned after rejuvenation.
      var allParticles = this.allParticles();
      assert(this.particlesAreInSync(allParticles));
      return resampleParticles(allParticles, function(resampledParticles){
        assert.strictEqual(resampledParticles.length, env.coroutine.numParticles);

        var numActiveParticles = _.reduce(resampledParticles, function(acc, p) {
          return acc + (p.trace.isComplete() ? 0 : 1);
        }, 0);

        if (numActiveParticles > 0) {
          // We still have active particles, wrap-around:
          env.coroutine.particleIndex = 0;
          return env.coroutine.rejuvenateParticles(resampledParticles, function(rejuvenatedParticles) {
            assert(env.coroutine.particlesAreInSync(rejuvenatedParticles));

            var p = _.partition(rejuvenatedParticles, function(p) { return p.trace.isComplete(); });
            env.coroutine.completeParticles = p[0];
            env.coroutine.particles = p[1];
            env.coroutine.debugLog(p[1].length + ' active particles after resample/rejuv.\n');

            if (env.coroutine.particles.length > 0) {
              return env.coroutine.runCurrentParticle();
            } else {
              return env.coroutine.finish();
            }
          }.bind(env.coroutine));
        } else {
          // All particles complete.
          env.coroutine.particles = [];
          env.coroutine.completeParticles = resampledParticles;
          return env.coroutine.finish();
        }
      });
    }
  };
```
It seems to cause error in util.js at `t = t()`, unlike `getSupport`.